curator-x-async/src/main/java/org/apache/curator/x/async/api/AsyncCuratorFrameworkDsl.java

```diff
@@ -112,4 +112,11 @@
      * @return builder object
      */
     AsyncRemoveWatchesBuilder removeWatches();
+
+    /**
+     * Start an add watch builder
+     *
```

Comment:
Consider adding something like: supported only when ZooKeeper client of version 3.6 or above is used, throws IllegalStateException for ZooKeeper client 3.5 or below